client/client.go

```diff
@@ -81,6 +85,17 @@ type Client struct {
 	manualOverride bool
 }
 
+func redirectError(req *http.Request, via []*http.Request) error {
```

Comment:
Could you please export this function so that we can use it in the cli?

Right now we build a client in `cli/command/cli.go`, so we wouldn't get this fix.

I also opened https://github.com/docker/cli/issues/64 to track this.